### ISSUE TITLE
infinite loop updating issue

### ISSUE THREAD

**Issue by oojacoboo** [nfl/react-helmet#194]
I have a couple vendor scripts that inject additional scripts into the `<head>`.  As a result of this, `react-helmet` goes into an infinite loop trying to update the head.  As a result of this, it's duplicating many of the scripts being inserted into the `<head>` and causing tons of errors, not to mention the obvious performance issues.

How can I prevent react-helmet from trying to modify the `<head>` after initially rendering.  I'm sure there is an event I can patch into and prefer propagation.

**Comment by oojacoboo** [comment]
Looks like the issue is related to the following:

```
module.exports = {
  target: isProd() ? 'node' : 'web',
  devtool: 'eval-source-map',
  devServer: {
    outputPath: root('dist'),
    colors: true,
    historyApiFallback: true, //set as false
    progress: true,
  },
```

I'm not sure exactly why the history API would be updating.  I'm guessing `react-helmet` is watching for pushState changes and trying to rebuild the `<head>` each time, omitting/merging in the injected script tags.

If there are any suggestions to improve/bypass/avoid this from happening, that'd be great.  Thanks.

**Comment by cwelch5** [comment]
@oojacoboo Helmet updates the head for every component in the route tree.  It definitely should not update infinitely.  If you have a very deep component tree it may do a lot of updates, which is a different issue.  Any vendor scripts not tagged with `data-react-helmet` will be left alone.

Are you still seeing this issue?  Sorry for the late response.

**Comment by oojacoboo** [comment]
Hey Chris, I'm not seeing the issue anymore.  The reason for the issue is because it seems anytime the head is modified by an external library it triggers Helmet to rebuild.  I saw this with libraries like Facebook and Google Maps.

We shifted some things around and was able to avoid that from happening though by managing everything through Helmet.  It happens when Helmet isn't actively managing everything in the head, at least from what I recall seeing.
